tests/startup_capture_test.go

```diff
@@ -0,0 +1,270 @@
+			{"ip netns exec fh-vm1 ip address add 169.254.33.33/24 dev fh-intf1", true},
+			{"ip netns exec fh-vm1 ip link set fh-intf1 up", true},
+
+			{"ovs-vsctl add-port br-startup fh-intf2 -- set interface fh-intf2 type=internal", true},
+			{"ip netns add fh-vm2", true},
+			{"ip link set fh-intf2 netns fh-vm2", true},
+			{"ip netns exec fh-vm2 ip address add 169.254.33.34/24 dev fh-intf2", true},
+			{"ip netns exec fh-vm2 ip link set fh-intf2 up", true},
+		},
+
+		settleFunction: func(c *TestContext) error {
+			// check that src and dst interfaces are in the right place before doing the ping
+			gremlin := `G.V().Has("Name", "fh-vm1").Out().Has("Name", "fh-intf1")`
+			nodes, err := c.gh.GetNodes(gremlin)
+			if err != nil || len(nodes) == 0 {
+				return errors.New("fh-intf1 not found in the expected namespace")
+			}
+
+			gremlin = `G.V().Has("Name", "fh-vm2").Out().Has("Name", "fh-intf2")`
+			nodes, err = c.gh.GetNodes(gremlin)
+			if err != nil || len(nodes) == 0 {
+				return errors.New("fh-intf2 not found in the expected namespace")
+			}
+
+			return nil
+		},
+
+		setupFunction: func(c *TestContext) error {
+			helper.ExecCmds(t, helper.Cmd{Cmd: "ip netns exec fh-vm1 ping -c 1 -s 1024 169.254.33.34", Check: false})
+			return nil
+		},
+
+		tearDownCmds: []helper.Cmd{
+			{"ip netns del fh-vm1", true},
+			{"ip netns del fh-vm2", true},
+			{"ovs-vsctl del-br br-startup", true},
+		},
+
+		// since the agent update ticker is about 10 sec according to the configuration
+		// we should wait 11 sec to have the first update and the MetricRange filled
+		checks: []CheckFunction{func(c *CheckContext) error {
+			prefix := "g"
+			if !c.time.IsZero() {
+				prefix += fmt.Sprintf(".Context(%d)", common.UnixMillis(c.time))
+			}
+			prefix += `.V().Has("Name", "br-startup", "Type", "ovsbridge")`
+
+			gh := c.gh
+			gremlin := prefix + `.Flows().Has("LayersPath", "Ethernet/IPv4/ICMPv4")`
+			flows, err := gh.GetFlows(gremlin)
+			if err != nil {
+				return err
+			}
+
+			if len(flows) != 1 {
+				return errors.New("We should receive only one ICMPv4 flow")
+			}
+
+			gremlin = fmt.Sprintf(prefix+`.Flows().Has("TrackingID", "%s").Nodes()`, flows[0].TrackingID)
+			tnodes, err := gh.GetNodes(gremlin)
+			if err != nil {
+				return err
+			}
+
+			if len(tnodes) != 3 {
+				return fmt.Errorf("We should have 3 nodes NodeTID,A,B got : %v for flows : %v", tnodes, flows)
+			}
+
+			gremlin = prefix + `.Flows().Has("LayersPath", "Ethernet/IPv4/ICMPv4").Hops()`
+			nodes, err := gh.GetNodes(gremlin)
+			if err != nil {
+				return err
+			}
+
+			if len(nodes) != 1 {
+				return errors.New("We should have 1 node NodeTID")
+			}
+
+			found := false
+			m := nodes[0].Metadata()
+			for _, n := range tnodes {
+				if n.MatchMetadata(m) == true {
+					found = true
+					break
+				}
+			}
+			if !found {
+				return errors.New("We should found the Hops nodes in the TrackingID nodes")
+			}
+
+			return nil
+		}},
+	}
+
+	RunStartupTest(t, test)
```

Comment:
Why copying the original RunTest and not using it ? did you do any changes, any feature is missing ?

Reply:
yes. they use list of captures provided by the test caller and create via client new captures. I dont want test to create new captures, but see that analyzer uses the pre-configured.
Of cause I can duplicate less code, but for tests I dont see if it is really needed

Reply:
Why not providing an empty list so that no captures will be created

Reply:
But then it also doesn't get the needed captures. I tried to do it. It doesnt work

Reply:
So it seems to me that it is more related to your patch, the capture should start without changing anything, the analyzer/agent are not started by RunTest but by init(). I'll have a look